libraries/chain/block_state.cpp

```diff
@@ -16,7 +17,29 @@ block_state::block_state(const block_header_state& bhs, deque<transaction_metada
    , block(std::make_shared<signed_block>(signed_block_header{bhs.header})) // [greg todo] do we need signatures?
    , pub_keys_recovered(true) // probably not needed
    , cached_trxs(std::move(trx_metas))
-{}
+{
+   block->transactions = std::move(trx_receipts);
+}
+
+// Used for transition from dbpos to instant-finality
+block_state::block_state(const block_state_legacy& bsp) {
+   block_header_state::id = bsp.id();
+   header = bsp.header;
+   activated_protocol_features = bsp.activated_protocol_features;
+   std::optional<block_header_extension> ext = bsp.block->extract_header_extension(instant_finality_extension::extension_id());
+   assert(ext); // required by current transistion mechanism
```

Comment:
```suggestion
   assert(ext); // required by current transition mechanism
```

Reply:
Hmm, I wonder if my editor last update broke the spell-checker. It is not flagging that.

Reply:
Fixed